pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -125,38 +100,48 @@ def __init__(
 
         self.monitor_op, self.kth_value, self.mode = mode_dict[mode]
 
-    def _del_model(self, filepath: str) -> None:
-        # shutil.rmtree(filepath)
-        os.remove(filepath)
+    def _del_model(self, filepath):
+        try:
+            shutil.rmtree(filepath)
+        except OSError:
+            os.remove(filepath)
 
-    def _save_model(self, filepath: str) -> None:
+    def _save_model(self, filepath):
         # make paths
-        os.makedirs(self.dirpath, exist_ok=True)
+        os.makedirs(os.path.dirname(filepath), exist_ok=True)
 
         # delegate the saving to the model
         if self.save_function is not None:
             self.save_function(filepath)
         else:
-            raise ValueError("Method `.save_function()` not set")
+            raise ValueError(".save_function() not set")
 
-    def check_monitor_top_k(self, current: float) -> bool:
+    def check_monitor_top_k(self, current):
         less_than_k_models = len(self.best_k_models) < self.save_top_k
         if less_than_k_models:
             return True
         return self.monitor_op(current, self.best_k_models[self.kth_best_model])
 
-    def _get_available_filepath(self, current: float, epoch: int) -> str:
-        current_str = f'{current:.2f}' if current else 'NaN'
-        fname = f'{self.prefix}_epoch={epoch}_{self.monitor}={current_str}'
-        filepath = os.path.join(self.dirpath, fname + self.EXTENSION)
-        assert not os.path.isfile(filepath)
-        return filepath
+    def format_checkpoint_name(self, epoch, logs):
```

Comment:
@Borda here

Reply:
nope

Reply:
this is  not right... already explained the goal. It's not about val_loss in particular. since this seems confusing, here's an example:

```
filepath='some/path/here/{spaceship_width:.2f}-{no_loss:.2f}-{epoch}'
produces: 
some/path/here/spaceship_width=0.20_no_loss=0.3_epoch=2.ckpt
```